decidim-core/app/commands/decidim/leave_user_group.rb

```diff
@@ -35,7 +36,13 @@ def leave_user_group
     end
 
     def can_leave?
-      Decidim::UserGroupMembership.where(user: user, user_group: user_group).where.not(role: :creator).any?
+      Decidim::UserGroupMembership.where(user: user, user_group: user_group).any?
+    end
+
+    def last_admin?
+      return false if Decidim::UserGroupMembership.where(user_group: user_group, user: user, role: [:creator, :admin]).empty?
```

Comment:
I think we don't need to do almost the same query twice here.

I think we can achieve the same result with a single query with something like:

```ruby
admin_members = Decidim::UserGroupMembership.where(user_group: user_group, role: [:creator, :admin])
admin_members.length == 1 && admin_members.pluck(:decidim_user_id).include?(user.id)
```